File: solutions/go/lasagna-master/1/lasagna_master.go
package lasagna

// TODO: define the 'PreparationTime()' function
func PreparationTime(layers []string, time int) int {
    if time == 0 {
        time = 2
    }
    return len(layers) * time
}

// TODO: define the 'Quantities()' function
func Quantities(layers []string) (int, float64) {
    var noodles int
    var sauce float64

    for _, layer := range layers {
        switch layer {
            case "noodles":
            	noodles += 50
            case "sauce":
            	sauce += 0.2
        }
    }

    return noodles, sauce
}

// TODO: define the 'AddSecretIngredient()' function
func AddSecretIngredient(friendsList []string, myList []string) {
    myList[len(myList)-1] = friendsList[len(friendsList)-1]
}

// TODO: define the 'ScaleRecipe()' function
func ScaleRecipe(amounts []float64, portions int) []float64 {
    var result []float64
    
    for _, amount := range amounts {
        result = append(result, amount * float64(portions)/2)
    }

    return result
}

// Your first steps could be to read through the tasks, and create
// these functions with their correct parameter lists and return types.
// The function body only needs to contain `panic("")`.
//
// This will make the tests compile, but they will fail.
// You can then implement the function logic one by one and see
// an increasing number of tests passing as you implement more
// functionality.

File: solutions/javascript/lasagna-master/1/lasagna-master.js
/// <reference path="./global.d.ts" />
// @ts-check

/**
 * Implement the functions needed to solve the exercise here.
 * Do not forget to export them so they are available for the
 * tests. Here an example of the syntax as reminder:
 *
 * export function yourFunction(...) {
 *   ...
 * }
 */

export function cookingStatus(remainingTime) {
  if (remainingTime === undefined) {
    return 'You forgot to set the timer.'
  }

  if (remainingTime === 0) {
    return 'Lasagna is done.'
  }

  return 'Not done, please wait.'
}

export function preparationTime(layers, averageTime = 2) {
  return layers.length * averageTime
}

export function quantities(layers) {
  const result = {
    'noodles': 0,
    'sauce': 0
  }

  for (let layer of layers) {
    switch (layer) {
      case 'noodles':
        result.noodles += 50
        break
      case 'sauce':
        result.sauce += 0.2
        break
    }
  }

  return result
}

export function addSecretIngredient(friendsList, myList) {
  myList.push(friendsList.at(-1))
}

export function scaleRecipe(recipe, quantities) {
  const result = {}
  
  for (let ingridient in recipe) {
    result[ingridient] = recipe[ingridient] * (quantities / 2)
  }
  
  return result
}

